source/static/js/download.js

```diff
@@ -0,0 +1,26 @@
+;
+(function(d, n) {
+  var os = n.platform.match(/(Win|Mac|Linux)/);
+  var x = n.userAgent.match(/x86_64|Win64|WOW64/) ||
+    n.cpuClass === 'x64' ? 'x64' : 'x86';
+  var base = 'https://iojs.org/dist/latest/';
+  var db = d.getElementById('home-downloadbutton');
+  var version = db.dataset.version;
+  var dlLocal = db.dataset.dlLocal;
+  var href;
+  switch (os && os[1]) {
+    case 'Mac':
+      href = 'iojs-' + version + '.pkg';
+      db.innerHTML = dlLocal + ' OS X (x64)';
+      break;
+    case 'Win':
+      href = 'iojs-' + version + '-' + x + '.msi';
+      db.innerHTML = dlLocal + ' Windows ('+x+')';
+      break;
+    case 'Linux':
+      href = 'iojs-' + version + '-linux-' + x + '.tar.gz';
+      db.innerHTML = dlLocal + ' Linux ('+x+')';
+      break;
+  }
+  db.href = base + href;
+})(document, navigator);
```

Comment:
Point 1: If you don't use javascript, it is just a button directly to https://iojs.org/dist/latest/ and show `Download` as usual.
Point 2: Is it really worth writing some tests?
Point 3: A default value as '' should be ok.
Point 4: Not a very usual thing.
Point 5: Each coin has two sides. That's why having a extra link below the button. If they find that our JS mis-detected their platform, they would directly click `Other Downloads`.